plugins/wazuh-core/public/services/http/server-client.ts

```diff
@@ -0,0 +1,513 @@
+/*
+ * Wazuh app - API request service
+ * Copyright (C) 2015-2024 Wazuh, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Find more information about this on the LICENSE file.
+ */
+import jwtDecode from 'jwt-decode';
+import { BehaviorSubject } from 'rxjs';
+import { Logger } from '../../../common/services/configuration';
+import {
+  HTTPClientServer,
+  HTTPVerb,
+  HTTPClientServerUserData,
+  WzRequestServices,
+  ServerAPIResponseItemsDataHTTPClient,
+} from './types';
+import { pluginPlatformRequestHeaders } from './constants';
+
+export class WzRequest implements HTTPClientServer {
+  onErrorInterceptor?: (
+    error: any,
+    options: {
+      checkCurrentApiIsUp: boolean;
+      shouldRetry: boolean;
+      overwriteHeaders?: any;
+    },
+  ) => Promise<void>;
+  private userData: HTTPClientServerUserData;
+  userData$: BehaviorSubject<HTTPClientServerUserData>;
+
+  constructor(
+    private readonly logger: Logger,
+    private readonly services: WzRequestServices,
+  ) {
+    this.userData = {
+      logged: false,
+      token: null,
+      account: null,
+      policies: null,
+    };
+    this.userData$ = new BehaviorSubject(this.userData);
+  }
+
+  /**
+   * Permorn a generic request
+   * @param {String} method
+   * @param {String} path
+   * @param {Object} payload
+   */
+  private async requestInternal(
+    method: HTTPVerb,
+    path: string,
+    payload: any = null,
+    // eslint-disable-next-line unicorn/no-object-as-default-parameter
+    extraOptions: {
+      shouldRetry?: boolean;
+      checkCurrentApiIsUp?: boolean;
+      overwriteHeaders?: any;
+    } = {
+      shouldRetry: true,
+      checkCurrentApiIsUp: true,
+      overwriteHeaders: {},
+    },
```

Comment:
Default parameters should not be passed to a function through an object literal. The `foo = {a: false}` parameter works fine if only used with one option. As soon as additional options are added, you risk replacing the whole `foo = {a: false, b: true}` object when passing only one option: `{a: true}`. For this reason, object destructuring should be used instead.

Reply:
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-object-as-default-parameter.md

Reply:
Fixed